sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -196,11 +222,13 @@ def do_resolve(value: Any, perform_apply: bool, exn: Optional[Exception]):
                 additionalSecretOutputs=additional_secret_outputs,
             )
 
+            from ..resource import create_urn
```

Comment:
I’ll admit - something seems wrong having all these local imports. I haven’t wrapped my head around what the core circularity issue is (and why it is different than the equivalent TyoeScript). But feels like this should not be the best we can do.

Reply:
Fixed.